command/agent/config.go

```diff
@@ -1371,6 +1376,9 @@ func MergeConfig(a, b *Config) *Config {
 	if b.NodeID != "" {
 		result.NodeID = b.NodeID
 	}
+	if b.DisableHostNodeID == true {
+		result.DisableHostNodeID = b.DisableHostNodeID
```

Comment:
Been working on a little framework to do everything with pointers so we can always tell if the user set it and merge cleanly, but haven't got everything ported to that yet - https://github.com/hashicorp/consul/blob/master/command/base/config_util.go.